### PR TITLE
Add release notes for the workbench

### LINKED ISSUE #25054: Workbench release notes

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Adds a set of release notes for the workbench and slightly reorganises the initial index page layout.

**To test:**

Build and read the updated documents.

Fixes #25054  

*This does not require release notes* because **this is the release notes.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v4.0.0/index.rst [hunk 1]
```diff
@@ -26,18 +26,32 @@ Changes
 -------
```
Comment (Mantid-Matthew) [inline docs/source/release/v4.0.0/index.rst]: Higher up in the this contains a line `Here is a load of stuff we have done.` and does not have the usual introductory paragraph/picture the release notes normally have. I think this is probably out of scope for this PR but just wanted to check.

File: docs/source/workbench/workspacehistorywindow.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+---------
+
+The scripting section allows a Python script to be generated from the history of the workspace. "Script To File" will request
+a name and directory for where to save the script whereas "Script to Clipboard" will copy the script to the system clipboard.
+
+When generating the script there are several options undre "Algorithm Versions" that control how the calls to algorithms will
+look in the generated script:
+
+- *Only specify Old Versions*: will cause a ``Version=`` argument to be given when calling an algorithm in the cases where the
+  ersion of the algorithm executed is older than the current highest version
```
Comment (Mantid-Matthew) [inline docs/source/workbench/workspacehistorywindow.rst]: missing a v from front of version here

Comment (martyngigg) [reply docs/source/workbench/workspacehistorywindow.rst]: :+1:

File: docs/source/release/v4.0.0/index.rst (hunk 1 above)
Comment (martyngigg) [reply docs/source/release/v4.0.0/index.rst]: I have a note on the project board to fill this in before the release is finalized. I think I'd prefer to do that in a separate issue if that's okay?

Comment (Mantid-Matthew) [reply docs/source/release/v4.0.0/index.rst]: That's fine by me just wanted to double check it didn't slip through the cracks

**Review by Matthew-Andrew — DISMISSED** [review]
LGTM

**Comment by martyngigg** [comment]
@Mantid-Matthew I've fixed a conflict. Could you re-approve?

**Pull request merged**